src/devices/Gpio/Drivers/Sunxi/SunxiDriver.cs

```diff
@@ -0,0 +1,476 @@
+                    return;
+                }
+
+                int fileDescriptor = Interop.open(GpioMemoryFilePath, FileOpenFlags.O_RDWR | FileOpenFlags.O_SYNC);
+                if (fileDescriptor == -1)
+                {
+                    throw new IOException($"Error {Marshal.GetLastWin32Error()} initializing the Gpio driver.");
+                }
+
+                IntPtr mapPointer0 = Interop.mmap(IntPtr.Zero, Environment.SystemPageSize - 1, (MemoryMappedProtections.PROT_READ | MemoryMappedProtections.PROT_WRITE), MemoryMappedFlags.MAP_SHARED, fileDescriptor, CpuxPortBaseAddess & ~_mapMask);
+                IntPtr mapPointer1 = Interop.mmap(IntPtr.Zero, Environment.SystemPageSize - 1, (MemoryMappedProtections.PROT_READ | MemoryMappedProtections.PROT_WRITE), MemoryMappedFlags.MAP_SHARED, fileDescriptor, CpusPortBaseAddess & ~_mapMask);
+                if (mapPointer0.ToInt64() == -1 || mapPointer1.ToInt64() == -1)
+                {
+                    throw new IOException($"Error {Marshal.GetLastWin32Error()} initializing the Gpio driver.");
+                }
+
+                Interop.close(fileDescriptor);
+
+                _gpioPointer0 = mapPointer0;
+                _gpioPointer1 = mapPointer0;
+
+                InitializeInterruptDriver();
+            }
+        }
+
+        /// <summary>
+        /// Map pin number with port controller name to pin number in the driver's logical numbering scheme.
+        /// </summary>
+        /// <param name="portController">Port controller name, like 'A', 'C'.</param>
+        /// <param name="port">Number of pins.</param>
+        /// <returns>Pin number in the driver's logical numbering scheme.</returns>
+        public static int MapPinNumber(char portController, int port)
+        {
+            int alphabetPosition = MapPortController(portController);
+
+            return alphabetPosition * 32 + port;
+        }
+
+        private (int PortController, int Port) UnmapPinNumber(int pinNumber)
+        {
+            int port = pinNumber % 32;
+            int portController = (pinNumber - port) / 32;
+
+            return (portController, port);
+        }
+
+        private static int MapPortController(char portController)
+        {
+            return portController switch
+            {
+                'A' => 0,
+                'B' => 1,
+                'C' => 2,
+                'D' => 3,
+                'E' => 4,
+                'F' => 5,
+                'G' => 6,
+                'H' => 7,
+                'I' => 8,
+                'J' => 9,
+                'K' => 10,
+                'L' => 11,
+                'M' => 12,
+                _ => throw new Exception()
+            };
+        }
+
+        private class PinState
+        {
+            public PinState(PinMode currentMode)
+            {
+                CurrentPinMode = currentMode;
+                InUseByInterruptDriver = false;
+            }
+
+            public PinMode CurrentPinMode
```

Comment:
please put it in one line (suggestion won't remove old lines)
```suggestion
            public PinMode CurrentPinMode { get; set; }
```

Reply:
This also needs to be changed...... https://github.com/dotnet/iot/blob/ddeecef97ebf6eb2901b169c46a02af47bd234eb/src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3LinuxDriver.cs#L641